### DISCUSSION TITLE
Improvements to the recognition algorithm

### DISCUSSION THREAD

**Discussion by robinreckmann** [puffinsoft/jscanify#23] [Ideas]
Hi,

thank you for this amazing project. I used it as a base to further improve the recognition algorithm as follows:

- Custom compiled opencv.js that only contains the functions we need => reduced file size from ~9 MB to about 1.8 MB.
- For paper extraction: Use OpenCV's grabCut algorithm to remove the background for higher precision when using the _extractPaper_ function (=> not used in realtime highlighting as the algorithm needs a few seconds to complete).
- Use morphologyEx to remove noise and text inside the documents.
- Use Canny edge detection to detect the boundaries of objects.
- Add cv.dilate to account for small inaccuracies in the Canny edge detection.
- Add a white border around the image to be able to detect documents not completely contained inside the canvas.
- Removed thresholding since its replaced by Canny edge and morphologyEx

I would just like to share it, maybe it is useful for other people as well. Maybe you can test it as well and consider adding some changes to your project.

[jscanify_modified.zip](https://github.com/ColonelParrot/jscanify/files/13331863/jscanify_modified.zip)

**Comment by thasneem077** [comment]
I am waiting....

**Comment by datrmine** [comment]
Would love to test, but link is broken...

**Comment by BjoernRave** [comment]
Would also be interested in these additions

**Comment by robinreckmann** [comment]
Attached you will find the files again. I added a wrapper script and some CSS, basically all you need to do to get this running is open the index.html file in a browser of your choice.

[jscanify_modified.zip](https://github.com/user-attachments/files/18166592/jscanify_modified.zip)

**Reply by ColonelParrot** [reply]
Apologies - I will take a look at it soon.

**Reply by BjoernRave** [reply]
yeah, would be super cool if the improvements could be merged back into the repo

**Comment by ColonelParrot** [comment]
Hello,
Thanks for taking the time to do this. It means a lot.
Now that I have some time, I've incorporated your suggestions as well as done some optimization on my end. Expect v1.3.0 to be released within the next day or two. It will have a much more robust recognition algorithm.
The new version should have the exact same API.

**Reply by ColonelParrot** [reply]
A larger test image dataset was used this time.

**Reply by robinreckmann** [reply]
Thanks a lot, I am looking forward to it!
